README.md

```diff
@@ -1,13 +1,48 @@
+
+There are three reasons to extend a project system in Visual Studio:
+1. Support a new project file format.
+1. Integrate existing file format with a new language service.
+1. Customize behavior of an existing project system.
+
+## Project system extensibility
+The traditional way to a build or customize a project system  is to implement a set of [Project System COM interfaces](https://docs.microsoft.com/en-us/visualstudio/extensibility/creating-a-basic-project-system-part-1). Most project systems don't start from scratch. Instead, they leverage the [MPFProj project system example](https://docs.microsoft.com/en-us/visualstudio/extensibility/internals/using-the-managed-package-framework-to-implement-a-project-type-csharp?view=vs-2017) as a starting point.
+
+The Visual Studio Project System (VSPS) described in this repository provides default implementation for a subset of project system COM interfaces. This simplifies building and maintaining a project system, but comes with the cost of reduced functionality.
+
+## How to select a project system platform?
+|Scenario|Recommended Project System Platform
+|---|---
+|Customize C#/VB/F# Desktop project system| COM-based [project flavoring](https://docs.microsoft.com/en-us/visualstudio/extensibility/internals/project-types?view=vs-2017).
+|Customize  C#/VB/F# .NET Core project system| Not fully supported at this time. VSPS is your best bet.
+|Add new a project type or a language| Prototype with VSPS, but be ready to fall back to MPFProj for complex scenarios that VSPS doesn't support yet.
+
+## Compare MPFProj and Visual Studio Project System
+|Criteria|[MPFProj](https://docs.microsoft.com/en-us/visualstudio/extensibility/internals/using-the-managed-package-framework-to-implement-a-project-type-csharp?view=vs-2017)| Visual Studio Project System (this repo)
+|---|---|---
+|Maturity|Release|Preview|
+|Breaking changes in major updates|None|[Expected](changes)|
+|Flexibility|Full control over project system behavior.|Some project system interfaces are not implemented.
+|Complexity|High: You will need to understand all project interfaces.|Low: You only need to understand customized behavior.
+|Extensibility|Implements project system COM interfaces.|C# with managed interfaces for extension points and MEF.
+|Threading model|Single threaded and bound to the UI thread.|Multi-threaded, scalable, and responsive.
+|Scalability|Memory-optimized|Uses extra memory to support multi-threading.
+|Methodology|Provides templates that implement project interfaces. The project system can modify or add to these templates.|Locks down project system interfaces and allows to override some behaviors.
+|Used by project systems|Desktop C#/VB/F#|[.NET Core C#/VB/F#](https://github.com/dotnet/project-system) and C++
+|Minimum Supported Version| Visual Studio 2008 | Visual Studio 2015
```

Comment:
Minimul Supported Versions -VS 2015: perhaps we should add a note that some features may be missing from older versions and that there are breaking changes between versions (with link to the breaking change page)